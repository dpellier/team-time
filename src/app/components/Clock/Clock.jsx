
import React from 'react';
import PropTypes from 'prop-types';
import { formatTime } from '../../utils/time';
import { TIME_FORMAT } from '../../utils/TIME_FORMAT';

const Clock = ({ time, timezone }) =>
    <div className='clock'>
        { formatTime(time, timezone, TIME_FORMAT.FULL_HOUR) }
    </div>
;

Clock.propTypes = {
    time: PropTypes.number,
    timezone: PropTypes.string
};

export { Clock };
