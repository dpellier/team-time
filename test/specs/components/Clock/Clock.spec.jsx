
import moment from 'moment-timezone';
import React from 'react';
import renderer from 'react-test-renderer';
import { Clock } from '../../../../src/app/components/Clock/Clock';

describe('<Clock />', () => {
    const usTimezone = 'America/Los_Angeles';

    it('is correctly rendered', () => {
        moment.tz.setDefault(usTimezone);
        const mockTime = moment('12:24:36', 'HH:mm:ss', usTimezone).valueOf();

        const component = renderer.create(
            <Clock time={ mockTime }
                   timezone={ usTimezone } />
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});
