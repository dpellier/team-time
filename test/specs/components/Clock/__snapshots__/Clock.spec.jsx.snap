// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock /> is correctly rendered 1`] = `
<div
  className="clock"
>
  12:24:36
</div>
`;
